ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/VertxNonceRequestTransmitter.java

```diff
@@ -74,35 +67,43 @@ public BigInteger requestNonce(final JsonRpcRequest request) {
     }
   }
 
-  private CompletableFuture<BigInteger> getNonceFromWeb3Provider(
-      final JsonRpcRequest requestBody, final MultiMap headers) {
+  private CompletableFuture<BigInteger> getNonceFromWeb3Provider(final JsonRpcRequest requestBody) {
+    final CompletableFuture<BigInteger> result = new CompletableFuture<>();
 
-    requestBody.setId(new JsonRpcRequestId(nextId.getAndIncrement()));
+    // TODO(we don't actually want it to be "this" - but it'll do for now...
+    final VertxRequestTransmitter transmitter =
+        transmitterFactory.create(
+            new ResponseBodyHandler() {
+              @Override
+              public void handleResponseBody(
+                  Map<String, String> headers, int statusCode, String body) {
+                VertxNonceRequestTransmitter.this.handleResponseBody(
+                    headers, statusCode, body, result);
+              }
+
+              @Override
+              public void handleTransmissionFailure(int statusCode, Throwable t) {
+                result.completeExceptionally(t);
+              }
+            });
 
-    final CompletableFuture<BigInteger> result = new CompletableFuture<>();
+    requestBody.setId(new JsonRpcRequestId(nextId.getAndIncrement()));
 
-    final HttpClientRequest request =
-        client.request(
-            HttpMethod.POST,
-            downstreamPathCalculator.calculateDownstreamPath("/"),
-            response -> response.bodyHandler(responseBody -> handleResponse(responseBody, result)));
-
-    request.setTimeout(requestTimeout.toMillis());
-    request.headers().setAll(headers);
-    request.exceptionHandler(result::completeExceptionally);
-    request.headers().remove("Content-Length"); // created during 'end'.
-    request.setChunked(false);
-    request.end(Json.encode(requestBody));
+    // TOOD(tmm) - shouldn't be an empty map! (need to convert headers!)
```

Comment:
is that TODO to be handled in current PR or in future PR?

Reply:
removed